### ISSUE TITLE
`redundantFileprivate` false positive, causing compiler issues

### ISSUE THREAD

**Issue by rogerluan** [nicklockwood/SwiftFormat#1435]
## Description

Hey 👋 

`redundantFileprivate` transformed my `fileprivate init()` into `private init()`, and caused compiler issue, because a subclass needed to call `super.init()`. Sample code:

### Before

```swift
public class MyParentClass {
    fileprivate init() { }
}

private class MySubclass: MyParentClass {
    init(something: String) {
        print(something)
        super.init()
    }
}
```

### After

```swift
public class MyParentClass {
    private init() { }
}

private class MySubclass: MyParentClass {
    init(something: String) {
        print(something)
        super.init() // Error: 'MyParentClass' initializer is inaccessible due to 'private' protection level
    }
}
```

Probably an edge case, and I totally blame it on poor usage of file structure (declaring multiple private/fileprivate classes within the same file), but just wanted to report anyway 😅 

Cheers

**Comment by nicklockwood** [comment]
@rogerluan thanks for reporting this - and no, it's nothing wrong with your code, just a case I didn't consider. I'll get it fixed.

**Comment by nicklockwood** [comment]
@rogerluan fixed in 0.51.8
